template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/validation/SafetyValveValidator.java

```diff
@@ -0,0 +1,40 @@
+package com.sequenceiq.cloudbreak.cmtemplate.validation;
+
+import java.util.List;
+
+import org.springframework.stereotype.Component;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.dataformat.xml.XmlMapper;
+import com.sequenceiq.cloudbreak.template.model.SafetyValve;
+
+@Component
+public class SafetyValveValidator {
+    public static final List<String> NON_XML_FORMAT_SAFETY_VALVE = List.of("hue_service_safety_valve",
+            "spark-conf/spark-defaults.conf_client_config_safety_valve",
+            "impala_cmd_args_safety_valve",
+            "hdfs_client_env_safety_valve",
+            "mapreduce_client_env_safety_valve",
+            "spark3-conf/spark-defaults.conf_client_config_safety_valve",
+            "REGIONSERVER_role_env_safety_valve");
+
+    private XmlMapper xmlMapper = new XmlMapper();
+
+    public void validate(SafetyValve safetyValve) {
+        if (NON_XML_FORMAT_SAFETY_VALVE.contains(safetyValve.getName())) {
+            return;
+        }
+        String wrappedValue = getWrappedValue(safetyValve.getRawValue());
+        try {
+            xmlMapper.readValue(wrappedValue, SafetyValve.class);
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException(String.format("Format is invalid for {serviceType: %s, roleType: %s, safety valve: %s}", safetyValve.getServiceType(),
```

Comment:
This one is also pending:
https://github.com/hortonworks/cloudbreak/pull/14712#discussion_r1172465354